pkg/util/file_indexer.go

```diff
@@ -95,6 +151,12 @@ func RunIndexer(directory string, ignoreRules []string) (filesChanged []string,
 		return filesChanged, filesDeleted, err
 	}
 
+	// check for .gitignore file and add odo-file-index.json to .gitignore
```

Comment:
I'm a bit sceptical about putting this here. This piece of code would run each time a user does `odo push`. Should this be a part of the push action or create, that is, `odo create`? I feel it should be in `odo create`. WDYT @kadel ?

Reply:
@dharmit IMO this should be in ```odo push``` because ```odo-file-index.json``` will always creates after ```odo push``` and ofcourse the code operation comes into picture after creation of ```odo-file-index.json```. Lets wait how @kadel see it.

Reply:
both make sense. It depends when we are adding it to the `.gitignore`. If we add `odo-file-index.json` to `.gitignore` when component (localConfig) is created that all the function should be in `create` package. If we are adding it to `.gitignore` when the `odo-file-index.json` file is created for the first time it should be in the package that creates `odo-file-index.json` file

Reply:
It is really not necessary to run this more than once. And there is no harm in adding it to `.gitignore` even if the file doesn't exist yet. 
I think it makes sense to do in `odo create` when the component gets initialized and `config.yaml` is created.


We can have it in fileindex also, but than we have to make sure that it is getting called only once when the `odo-file-index.json` is being created, definitely not every time the file is updated.